src/redux/api/adminApi.tsx

```diff
@@ -0,0 +1,62 @@
+import { IAdmin, INewAdmin } from "@/interfaces/adminData";
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+import { environment } from "../../constants/environment";
+
+export const adminApi = createApi({
+  reducerPath: "admin",
+  baseQuery: fetchBaseQuery({ baseUrl: environment.apiBaseUrl }),
+  tagTypes: ["admin"],
+  endpoints: (builder) => ({
+    getAllAdmin: builder.query({
+      query: () => "/admin",
+      providesTags: [{ type: "admin", id: "LIST" }],
+    }),
+    getAdminById: builder.query<IAdmin, string>({
+      query: (adminId: string ) => `/admin/${adminId}`,
+      providesTags: (_1, _2, adminId) => [{ type: "admin", id: adminId }],
+    }),
+    updateAdmin: builder.mutation({
+      query: (admin: IAdmin) => ({
+        url: `/admin/${admin._id}`,
+        method: "PATCH",
+        body: admin,
+      }),
+      invalidatesTags: (_1, _2, admin) => [
+        { type: "admin", id: admin._id },
+        { type: "admin", id: "LIST" },
+      ],
+    }),
+    deleteAdmin: builder.mutation({
+      query: (adminId: string) => ({
+        url: `/admin/${adminId}`,
+        method: "DELETE",
+      }),
+      invalidatesTags: (_1, _2, adminId) => [
+        { type: "admin", id: adminId },
+        { type: "admin", id: "LIST" },
+      ],
+    }),
+    restoreAdmin: builder.mutation({
+      query: (adminId: string) => ({
+        url: `/admin/${adminId}/restore`,
+        method: "PATCH",
+      }),
+      invalidatesTags: (_1, _2, adminId) => [
+        { type: "admin", id: adminId },
+        { type: "admin", id: "LIST" },
+      ],
+    }),
+    createAdmin: builder.mutation({
```

Comment:
Currently, the update method in admin.service.ts at server side will report error if one admin is deleted because we want the update only working for active accounts. The already "isdeleted=true" account needs a different stream to handle.